### DISCUSSION TITLE
Invalidation from Mutations not working

### DISCUSSION THREAD

**Discussion by youneshajji** [TanStack/query#3317] [Q&A]
Hi,

I have implemented the Invalidation from Mutations but it doesnt seem to be working for me, when i invalidate from devtools manually it works fine. After i trigger a mutation and switch browser panes back and forth, i can see the data changes in the UI. What am i missing :) 

Here is my code:

  const userProfileQuery = useQuery(
    "userProfileData",
    () =>
      fetchApi(process.env.REACT_APP_API_URL_APIM + "/Contacts", account, inProgress, instance).then((res) => res.data),
    {
      onSuccess: (data) => {
        ...
      },
      onError: (_error) => {
        toast.error(_error.message, {
          position: "top-center",
        })
      },
    }
  )

const mutationUpdate = useMutation(
    () => {
      patchApi(
        process.env.REACT_APP_API_URL_APIM + "/Contacts/me",
        {
          ...
        },
        account,
        inProgress,
        instance
      )
    },
    {
      onSuccess: async () => {
        queryClient.invalidateQueries("userProfileData")
      },
      onError: (_error) => {
        toast.error(_error.message, {
          position: "top-center",
        })
      },
      onSettled: () => {
        hideShowSpans(false)
        onToggleEdit()
        toast.success("Profilen din er oppdatert!", {
          position: "top-center",
        })
      },
    }
  )

**Comment by TkDodo** [comment]
please try to provide that in a codesandbox reproduction
